authz/providers/azure/rbac/rbac.go

```diff
@@ -0,0 +1,198 @@
+	u := &AccessInfo{
+		client: http.DefaultClient,
+		headers: http.Header{
+			"Content-Type": []string{"application/json"},
+		},
+		apiURL:          rbacURL,
+		tokenProvider:   tokenProvider,
+		azureResourceId: resourceId}
+
+	if clsuterType == "arc" {
+		u.clusterType = CONNECTED_CLUSTER
+	}
+
+	if clsuterType == "aks" {
+		u.clusterType = MANAGED_CLUSTER
+	}
+
+	return u
+}
+
+func New(clientID, clientSecret, tenantID string, useGroupUID bool, aadEndpoint, armEndPoint, clusterType, resourceId string) *AccessInfo {
+	rbacURL, _ := url.Parse(armEndPoint)
+
+	tokenProvider := graph.NewClientCredentialTokenProvider(clientID, clientSecret,
+		fmt.Sprintf("%s%s/oauth2/v2.0/token", aadEndpoint, tenantID),
+		fmt.Sprintf("%s.default", armEndPoint))
+
+	return newAccessInfo(tokenProvider, rbacURL, useGroupUID, clusterType, resourceId)
+}
+
+func NewWithAKS(tokenURL, tenantID, armEndPoint, clusterType, resourceId string) *AccessInfo {
+	rbacURL, _ := url.Parse(armEndPoint)
+	tokenProvider := graph.NewAKSTokenProvider(tokenURL, tenantID)
+
+	return newAccessInfo(tokenProvider, rbacURL, true, clusterType, resourceId)
+}
+
+func (a *AccessInfo) RefreshToken() error {
+	resp, err := a.tokenProvider.Acquire("")
+	if err != nil {
+		glog.Errorf("%s failed to refresh token : %s", a.tokenProvider.Name(), err.Error())
+		return errors.Errorf("%s: failed to refresh token: %s", a.tokenProvider.Name(), err)
+	}
+
+	// Set the authorization headers for future requests
+	a.headers.Set("Authorization", fmt.Sprintf("Bearer %s", resp.Token))
+	expIn := time.Duration(resp.Expires) * time.Second
+	a.expires = time.Now().Add(expIn - expiryDelta)
+
+	return nil
+}
+
+func (a *AccessInfo) IsTokenExpired() bool {
+	if a.expires.Before(time.Now()) {
+		return true
+	} else {
+		return false
+	}
+}
+
+func (a *AccessInfo) CheckAccess(request *authzv1.SubjectAccessReviewSpec) (*authzv1.SubjectAccessReviewStatus, error) {
+	var CHECKACCESS_PATH string = "/providers/Microsoft.Authorization/checkaccess"
+	var API_VERSION string = "2018-09-01-preview"
+	checkAccessBody := PrepareCheckAccessRequest(request, a.clusterType, a.azureResourceId)
+	checkAccessURL := *a.apiURL
+	// Append the path for azure cluster resource id
+	checkAccessURL.Path = path.Join(checkAccessURL.Path, a.azureResourceId)
+	var str string
+	if getNameSpaceScope(request, &str) {
+		checkAccessURL.Path = path.Join(checkAccessURL.Path, str)
+	}
+
+	checkAccessURL.Path = path.Join(checkAccessURL.Path, CHECKACCESS_PATH)
+	params := url.Values{}
+	params.Add("api-version", API_VERSION)
+	checkAccessURL.RawQuery = params.Encode()
+
+	if a.IsTokenExpired() {
+		a.RefreshToken()
+	}
+
+	buf := new(bytes.Buffer)
+	if err := json.NewEncoder(buf).Encode(checkAccessBody); err != nil {
+		return nil, errors.Wrap(err, "error encoding check access request")
+	}
+
+	binaryData, _ := json.MarshalIndent(checkAccessBody, "", "    ")
```

Comment:
just use Marshal

Reply:
This prints data in readable format. Is there any issue with indent?